### PR TITLE
Add `fromfrac` function for converting fractions to dividable types

### LINKED ISSUE #1309: Add interface for converting `Fraction`s to other types
Currently there is no easy interface for turning fractions into floats (because of the understanding that they will lose information). 

Probably a method in either `Rational` or `Real` would suffice

### PR DESCRIPTION
**Pull request by Izaakwltn**
Closes #1309

### CODE REVIEW

**Comment by Izaakwltn** [comment]
@stylewarning is there a better name for this function?

**Review by jbouwman — APPROVED** [review]
approve, pending considering the function's name

**Comment by stylewarning** [comment]
fromfrac? a sort of analog of fromint?

maybe that can go into the docs too

File: library/math/real.lisp [hunk 1]
```diff
@@ -264,7 +265,14 @@ Note: This does *not* divide double-float arguments."
   (declare round/ (Integer -> Integer -> Integer))
   (define (round/ a b)
     "Divide two integers and round the quotient."
-    (round (exact/ a b))))
+    (round (exact/ a b)))
+
+  (declare fromfrac (Dividable Integer :a => Fraction -> :a))
+  (define (fromfrac q)
+    "Converts a fraction to a type `:a` with an instance of `Dividable Integer :a`, similar to `Num`'s `fromint`.
```
Comment (stylewarning) [inline library/math/real.lisp:272]:
```
Converts a `Fraction` to a target type. More specifically, ...
```

File: docs/intro-to-coalton.md [hunk 2]
```diff
@@ -546,6 +546,8 @@ All of these cases are sufficiently common that we provide a few shorthands:
 
 - `floor/`, `ceiling/`, and `round/` for integer-to-integer division, and
 
+- `fromfrac` for converting a `Fraction` into a type with a `Dividable Integer :a` instance (this conversion may result in precision loss)
```
Comment (stylewarning) [inline docs/intro-to-coalton.md:549]: I think this deserves an example, and maybe shouldn't be a bullet point, but be a separate paragraph altogether.

**Review by stylewarning — CHANGES_REQUESTED** [review]

File: docs/intro-to-coalton.md [hunk 3]
```diff
@@ -548,6 +548,16 @@ All of these cases are sufficiently common that we provide a few shorthands:
 
 - `floor/`, `ceiling/`, and `round/` for integer-to-integer division, and
 
+Fractions can be converted to other dividable types using `fromfrac` (Note: This may result in precision loss):
+
+```
+COALTON-LIBRARY/MATH/REAL> (coalton (the Double-Float (fromfrac (/ 1 2))))
```
Comment (stylewarning) [inline docs/intro-to-coalton.md:554]: you can just write 1/2 and 999/1000 and the like, right?

**Review by stylewarning — APPROVED** [review]

File: docs/intro-to-coalton.md (hunk 3 above)
Comment (Izaakwltn) [reply docs/intro-to-coalton.md]: Yup! changed them and checked them

**Review by stylewarning — APPROVED** [review]

**Pull request merged**